### PR TITLE
Delete wallet

### PR DESCRIPTION
**Pull request by lrubiorod**

### CODE REVIEW

**Review by tmpolaczyk — APPROVED** [review]

File: wallet/src/actors/app/methods.rs [hunk 1]
```diff
@@ -187,6 +187,27 @@ impl App {
         Box::pin(f)
     }
 
+    /// Delete wallet
```
Comment (mariocao) [inline wallet/src/actors/app/methods.rs:190]: I think you can document this better :rofl:

File: wallet/src/actors/worker/methods.rs [hunk 2]
```diff
@@ -232,6 +232,13 @@ impl Worker {
         Ok(id)
     }
 
+    /// Delete wallet
```
Comment (mariocao) [inline wallet/src/actors/worker/methods.rs:235]: Same here

File: wallet/src/repository/wallets/mod.rs [hunk 3]
```diff
@@ -122,6 +122,24 @@ impl<T: Database> Wallets<T> {
         Ok(())
     }
 
+    /// Delete a wallet by its id
```
Comment (mariocao) [inline wallet/src/repository/wallets/mod.rs:125]:
```suggestion
    /// Delete a wallet by its ID
```

File: wallet/src/repository/wallets/mod.rs [hunk 4]
```diff
@@ -122,6 +122,24 @@ impl<T: Database> Wallets<T> {
...
+    pub fn delete(&self, wallet_id: String) -> Result<()> {
+        let mut batch = self.db.batch();
+        // FIXME: Use merge operator or a transaction when available in rocksdb crate
```
Comment (mariocao) [inline wallet/src/repository/wallets/mod.rs:128]:
```suggestion
```

**Review by mariocao — CHANGES_REQUESTED** [review]
I just discovered that `delete_wallet` does not remove the wallet session. Therefore, other API calls are still allowed (e.g. `generate_address`) which does not make much sense. :smile:

**Review by mariocao — COMMENTED** [review]
I discovered another "strange" flow:

1. Create wallet
2. Unlock wallet -> session id
3. Delete wallet
4. Unlock wallet with `session_id` --> IT WORKS! :)

`unlock_wallet` does not check if the `wallet_id` exists in the list of known wallets.
It directly tries to decrypt the wallet from the database.

It is not an error per se, but it could feel strange for a user.

File: wallet/src/repository/wallets/mod.rs [hunk 5]
```diff
@@ -122,11 +122,33 @@ impl<T: Database> Wallets<T> {
...
+
+        Ok(())
+    }
+
     /// Get a wallet's salt and IV based on its provided ID
     pub fn wallet_salt_and_iv(&self, id: &str) -> Result<(Vec<u8>, Vec<u8>)> {
-        let salt = self.db.get(&keys::wallet_id_salt(id))?;
-        let iv = self.db.get(&keys::wallet_id_iv(id))?;
+        let ids: Vec<String> = self.db.get_or_default(&keys::wallet_ids())?;
+        if !ids.contains(&id.to_string()) {
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallets/mod.rs:145]: Add a comment, mention that this is needed because deleted otherwise wallets can still be unlocked.

**Review by mariocao — APPROVED** [review]

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**